demo/src/App.js

```diff
@@ -1,188 +1,183 @@
-      viewHeight: window.innerHeight * 0.9,
-      viewWidth: window.innerWidth * 0.7
+  const toggleChannel = index => {
+    setChannelsOn(prevChannelsOn => {
+      const nextChannelsOn = [...prevChannelsOn];
+      nextChannelsOn[index] = !nextChannelsOn[index];
+      return nextChannelsOn;
     });
-  }
+  };
 
-  render() {
-    const { sourceName } = this.state;
-    const initialViewState = sources[sourceName].initialViewState || {
-      zoom: -5.5,
-      target: [30000, 10000, 0]
-    };
-    const {
-      sliderValues,
-      colorValues,
-      viewHeight,
-      viewWidth,
-      channelsOn
-    } = this.state;
-    const sliders = Object.keys(this.sliders).map(channel => {
-      const ChannelSlider = this.sliders[channel];
-      const ChannelCheckbox = this.checkboxes[channel];
-      const sliderValue = sliderValues[channel];
+  const sourceButtons = Object.keys(sources).map(name => {
+    return (
+      <Button
+        variant="contained"
+        key={name}
+        disabled={name === sourceName}
+        onClick={() => setSourceName(name)}
+      >
+        {name}
+      </Button>
+    );
+  });
+
+  const sliders = Object.keys(sources[sourceName].channels).map(
+    (channel, i) => {
+      const [ColoredSlider, ColoredCheckbox] = styledSelectors[i];
       return (
         <div key={`container-${channel}`}>
           <p>{channel}</p>
           <div style={{ width: '100%', display: 'flex', position: 'relative' }}>
-            <ChannelCheckbox
-              // eslint-disable-next-line no-unused-vars
-              onChange={e => this.toggleChannel(channel)}
-              checked={channelsOn[channel]}
+            <ColoredCheckbox
+              onChange={() => toggleChannel(i)}
+              checked={channelsOn[i]}
             />
-            <ChannelSlider
+            <ColoredSlider
               style={{ top: '7px' }}
-              value={sliderValue}
-              onChange={(e, v) => this.handleSliderChange(e, v, channel)}
+              value={sliderValues[i]}
+              onChange={(event, value) => handleSliderChange(i, value)}
               valueLabelDisplay="auto"
               getAriaLabel={() => channel}
-              min={0}
-              max={this.max}
+              min={MIN_SLIDER_VALUE}
+              max={MAX_SLIDER_VALUE}
               orientation="horizontal"
             />
           </div>
         </div>
       );
-    });
+    }
+  );
 
-    const sourceButtons = Object.keys(sources).map(name => {
-      return (
-        <Button
-          variant="contained"
-          key={name}
-          disabled={name === sourceName}
-          onClick={() => {
-            this.setState({ sourceName: name });
-          }}
-        >
-          {name}
-        </Button>
-      );
-    });
+  const source = sources[sourceName];
 
-    const source = sources[sourceName];
-    return (
-      <div>
+  const initialViewState =
+    sources[sourceName].initialViewState || DEFAULT_VIEW_CONFIG;
```

Comment:
Maybe change the name to `DEFAULT_VIEW_STATE`... or change `*State` to `*Config`?